### PR TITLE
Adding Dogstatsd telemetry doc

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

Adding documentation about DogStatsd clients telemetry.

### Preview link

https://docs-staging.datadoghq.com/maxime/add-statsd-telemetry-doc/developers/dogstatsd/high_throughput

### CODE REVIEW

File: content/en/developers/dogstatsd/high_throughput.md [hunk 1]
```diff
@@ -212,6 +212,98 @@ Then set the Agent `dogstatsd_so_rcvbuf` configuration option to the same number
...
+| `datadog.dogstatsd.client.bytes_dropped`          | count        | Number of bytes dropped by the client.                                          |
+| `datadog.dogstatsd.client.bytes_dropped_queue`    | count        | Number of bytes dropped because the client queue was full.                      |
+| `datadog.dogstatsd.client.bytes_dropped_writer`   | count        | Number of bytes dropped because of an error while writing to the transport.     |
+| `datadog.dogstatsd.client.packets_sent`           | count        | Number of datagrams successfully sent to the agent.                             |
+| `datadog.dogstatsd.client.packets_dropped`        | count        | Number of datagrams dropped by the client.                                      |
+| `datadog.dogstatsd.client.packets_dropped_queue`  | count        | Number of datagrams dropped because the client queue was full.                  |
+| `datadog.dogstatsd.client.packets_dropped_writer` | count        | Number of datagrams dropped because of an error while writing to the transport. |
+
+
+If you want to disable telemetry, simply do (see [DataDog/datadog-go][1] for more information about the client configuration):
```
Comment (remeh) [inline content/en/developers/dogstatsd/high_throughput.md]: Extra-nit, I would go with something like:

```
If you want to disable telemetry, simply do:

'''go
statsd, err: = statsd.New("127.0.0.1:8125", statsd.WithoutTelemetry())
'''

See [DataDog/datadog-go](https://github.com/DataDog/datadog-go) for more information about the client configuration.
```

WDYT?

File: content/en/developers/dogstatsd/high_throughput.md [hunk 2]
```diff
@@ -212,6 +212,98 @@ Then set the Agent `dogstatsd_so_rcvbuf` configuration option to the same number
 dogstatsd_so_rcvbuf: 26214400
 ```
 
+## Client side telemetry
+
+Dogstatsd clients send telemetry metrics by default to the agent. This allows
+you to better troubleshoot where the bottleneck might be. Each metrics will be
```
Comment (remeh) [inline content/en/developers/dogstatsd/high_throughput.md]: typo: `Each metric will be tagged`

File: content/en/developers/dogstatsd/high_throughput.md [hunk 3]
```diff
@@ -212,6 +212,98 @@ Then set the Agent `dogstatsd_so_rcvbuf` configuration option to the same number
...
+and the corresponding metrics will not reflect bytes/packets drop.
+
+{{< tabs >}}
+{{% tab "Python" %}}
+
+Starting with version `0.34.0` of the Python client.
+
+| Metrics Name                               | Metric Type | Description                                                                               |
+|--------------------------------------------|-------------|-------------------------------------------------------------------------------------------|
+| `datadog.dogstatsd.client.metrics`         | count       | Number of `metrics` sent to the client by yout application (before sampling).             |
```
Comment (remeh) [inline content/en/developers/dogstatsd/high_throughput.md]: typo: `by your application`
on next 2 lines too.

File: content/en/developers/dogstatsd/high_throughput.md [hunk 4]
```diff
@@ -212,6 +212,98 @@ Then set the Agent `dogstatsd_so_rcvbuf` configuration option to the same number
...
+| `datadog.dogstatsd.client.event`           | count       | Number of `events` sent to the client by yout application.                                |
```
Comment (remeh) [inline content/en/developers/dogstatsd/high_throughput.md]: sanity-check: is the `event` missing a `s` in the metric name?

File: content/en/developers/dogstatsd/high_throughput.md (hunk 1 above)
Comment (hush-hush) [reply content/en/developers/dogstatsd/high_throughput.md]: much better :+1:

File: content/en/developers/dogstatsd/high_throughput.md (hunk 4 above)
Comment (hush-hush) [reply content/en/developers/dogstatsd/high_throughput.md]: not but it should ! I'll update the python and go PR. :ok_hand:

File: content/en/developers/dogstatsd/high_throughput.md [hunk 5]
```diff
@@ -212,6 +212,98 @@ Then set the Agent `dogstatsd_so_rcvbuf` configuration option to the same number
 dogstatsd_so_rcvbuf: 26214400
 ```
 
+## Client side telemetry
+
+Dogstatsd clients send telemetry metrics by default to the agent. This allows
+you to better troubleshoot where the bottleneck might be. Each metric will be
```
Comment (sarina-dd) [inline content/en/developers/dogstatsd/high_throughput.md]:
```suggestion
you to better troubleshoot where bottleneck exists. Each metric will be
```

File: content/en/developers/dogstatsd/high_throughput.md [hunk 6]
```diff
@@ -212,6 +212,98 @@ Then set the Agent `dogstatsd_so_rcvbuf` configuration option to the same number
...
+tagged with the client language and the client version. Those metrics will not be
```
Comment (sarina-dd) [inline content/en/developers/dogstatsd/high_throughput.md]:
```suggestion
tagged with the client language and the client version. These metrics will not be
```

File: content/en/developers/dogstatsd/high_throughput.md [hunk 7]
```diff
@@ -212,6 +212,98 @@ Then set the Agent `dogstatsd_so_rcvbuf` configuration option to the same number
...
+counted as custom and will not be billed.
```
Comment (sarina-dd) [inline content/en/developers/dogstatsd/high_throughput.md]:
```suggestion
counted as custom metrics and will not be billed.
```

File: content/en/developers/dogstatsd/high_throughput.md [hunk 8]
```diff
@@ -212,6 +212,98 @@ Then set the Agent `dogstatsd_so_rcvbuf` configuration option to the same number
...
+
+Each client share a set of common tags.
+
+| Tag                | Description                                     | Example                |
+|--------------------|-------------------------------------------------|------------------------|
+| `client`           | The language of the client                      | `client:py`            |
+| `client_version`   | The version of the client                       | `client_version:1.2.3` |
+| `client_transport` | The transport byte the client (`udp` or `uds`)  | `client_transport:uds` |
+
+Keep in mind that when using UDP, network errors can't be detected by the client
```
Comment (sarina-dd) [inline content/en/developers/dogstatsd/high_throughput.md]:
```suggestion
**Note**: When using UDP, network errors can't be detected by the client
```

File: content/en/developers/dogstatsd/high_throughput.md [hunk 9]
```diff
@@ -212,6 +212,98 @@ Then set the Agent `dogstatsd_so_rcvbuf` configuration option to the same number
...
+| `datadog.dogstatsd.client.metrics`         | count       | Number of `metrics` sent to the client by your application (before sampling).             |
+| `datadog.dogstatsd.client.events`          | count       | Number of `events` sent to the client by your application.                                |
+| `datadog.dogstatsd.client.service_checks`  | count       | Number of `service_checks` sent to the client by your application.                        |
+| `datadog.dogstatsd.client.bytes_sent`      | count       | Number of bytes successfully sent to the Agent.                                           |
+| `datadog.dogstatsd.client.bytes_dropped`   | count       | Number of bytes dropped by the client.                                                    |
+| `datadog.dogstatsd.client.packets_sent`    | count       | Number of datagrams successfully sent to the agent.                                       |
+| `datadog.dogstatsd.client.packets_dropped` | count       | Number of datagrams dropped by the client.                                                |
+
+If you want to disable telemetry, simply do:
```
Comment (sarina-dd) [inline content/en/developers/dogstatsd/high_throughput.md]:
```suggestion
If you want to disable telemetry:
```

File: content/en/developers/dogstatsd/high_throughput.md [hunk 10]
```diff
@@ -212,6 +212,98 @@ Then set the Agent `dogstatsd_so_rcvbuf` configuration option to the same number
...
+| `datadog.dogstatsd.client.bytes_dropped`          | count        | Number of bytes dropped by the client.                                          |
+| `datadog.dogstatsd.client.bytes_dropped_queue`    | count        | Number of bytes dropped because the client queue was full.                      |
+| `datadog.dogstatsd.client.bytes_dropped_writer`   | count        | Number of bytes dropped because of an error while writing to the transport.     |
+| `datadog.dogstatsd.client.packets_sent`           | count        | Number of datagrams successfully sent to the agent.                             |
+| `datadog.dogstatsd.client.packets_dropped`        | count        | Number of datagrams dropped by the client.                                      |
+| `datadog.dogstatsd.client.packets_dropped_queue`  | count        | Number of datagrams dropped because the client queue was full.                  |
+| `datadog.dogstatsd.client.packets_dropped_writer` | count        | Number of datagrams dropped because of an error while writing to the transport. |
+
+
+If you want to disable telemetry, simply do:
```
Comment (sarina-dd) [inline content/en/developers/dogstatsd/high_throughput.md]:
```suggestion
If you want to disable telemetry:
```

File: content/en/developers/dogstatsd/high_throughput.md [hunk 11]
```diff
@@ -212,6 +212,98 @@ Then set the Agent `dogstatsd_so_rcvbuf` configuration option to the same number
...
 ```
 
+## Client side telemetry
+
+Dogstatsd clients send telemetry metrics by default to the agent. This allows
+you to better troubleshoot where bottleneck exists. Each metric will be
+tagged with the client language and the client version. These metrics will not be
+counted as custom metrics and will not be billed.
+
+Each client share a set of common tags.
```
Comment (sarina-dd) [inline content/en/developers/dogstatsd/high_throughput.md]:
```suggestion
Each client shares a set of common tags.
```

**Pull request merged**